MinecraftClient/ChatBots/MineCube.cs

```diff
@@ -0,0 +1,340 @@
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX + 1, blockY, blockZ)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX, blockY, blockZ - 1)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX, blockY, blockZ + 1)).Type);
+        }
+
+        /// <summary>
+        /// Prints a whole cube to the console. Separated in layers and rows.
+        /// </summary>
+        /// <param name="cubeToPrint">Some cube</param>
+        private void PrintCubeToConsole(Cube cubeToPrint)
+        {
+            LogToConsole("Cube generated:");
+            foreach (Layer lay in cubeToPrint.LayersToMine)
+            {
+                LogToConsole("Layer:");
+                foreach (Row r in lay.RowsToMine)
+                {
+                    string generatedRow = "Row: ";
+                    foreach (Location loc in r.BlocksToMine)
+                    {
+                        generatedRow += loc.ToString() + "; ";
+                    }
+                    LogToConsole(generatedRow);
+                }
+            }
+            LogToConsole("End of cube.");
+        }
+
+        private string EvaluateMineUpCommand(string command, string[] args)
+        {
+            Location startBlock = new Location(
+                        double.Parse(args[0]),
+                        double.Parse(args[1]),
+                        double.Parse(args[2])
+                        );
+
+            Location stopBlock = new Location(
+                    double.Parse(args[3]),
+                    double.Parse(args[4]),
+                    double.Parse(args[5])
+                    );
+
+            if (Math.Round(startBlock.Y) != Math.Round(stopBlock.Y))
+            {
+                return "Both blocks must have the same Y value!";
+            }
+
+            CubeFromWorld CFW = new CubeFromWorld();
+            List<Material> materialWhitelist = new List<Material>(new Material[] { Material.Air });
+            Thread tempThread = new Thread(() => MineUp(CFW.GetBlocksAsCube(GetWorld(), startBlock, stopBlock, materialWhitelist, isBlacklist:false)));
+            tempThread.Start();
+
+            return "Start mining up.";
+        }
+
+        private string EvaluateMineCommand(string command, string[] args)
+        {
+            if (args.Length > 2)
+            {
+                Location startBlock;
+                Location stopBlock; 
+
+                if (args.Length > 5)
+                {
+                    startBlock = new Location(
+                    double.Parse(args[0]),
+                    double.Parse(args[1]),
+                    double.Parse(args[2])
+                    );
+
+                    stopBlock = new Location(
+                    double.Parse(args[3]),
+                    double.Parse(args[4]),
+                    double.Parse(args[5])
```

Comment:
Validate user input before using them. You may add a `try/catch` block to catch invalid input.